util/webhook/webhook.go

```diff
@@ -191,9 +203,32 @@ func affectedRevisionInfo(payloadIf interface{}) (webURLs []string, revision str
 		webURLs = append(webURLs, payload.Repository.Links.HTML.Href)
 		// TODO: bitbucket includes multiple changes as part of a single event.
 		// We only pick the first but need to consider how to handle multiple
-		for _, change := range payload.Push.Changes {
-			revision = change.New.Name
-			break
+		fullName := strings.Split(payload.Repository.FullName, "/")
+		workspace := fullName[0]
+		repoSlug := payload.Repository.Name
+
+		for _, changes := range payload.Push.Changes {
+			revision = changes.New.Name
+			change.shaBefore = changes.Old.Target.Hash
+			change.shaAfter = changes.New.Target.Hash
+
+			if webhookHandler != nil {
+				spec := change.shaAfter + ".." + change.shaBefore
+				accessToken, err := webhookHandler.settingsSrc.GetRepositorySecretAccessToken(webURLs[0])
```

Comment:
webURLs[0] doesn't read well here, i.e. what makes index 0 special?

Reply:
Pls have a look at the  [webhook payload](https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/?focusedCommentId=774045711#comment-774045711), you can see that the bitbucket includes multiple changes in a single event, so I have appended the URL of repo in webURL. So only one webURLs is present in case of bitbucket, whereas it is different for github, gitlab, etc. So it has to be a array, wherein only one item is present in case of bitbucket. That's why index 0 is used.